lib/gmail-adapter.js

```diff
@@ -34,8 +34,23 @@ var Read = Juttle.proc.base.extend({
     start: function() {
         var self = this;
 
+        self.options.from = self.options.from || new JuttleMoment(0);
+
+        return self.get_messages_for_timerange(self.options.from, self.options.to);
+    },
+
+    teardown: function() {
```

Comment:
teardown()? I see it in juttle/lib/runtime/procs/base.js, looks like it's supposed to handle local teardown of anything for a given proc. That contrasts with deactivate(), which recursively calls deactivate() for connected graph procs.